fairlearn/_input_validation.py

```diff
@@ -5,70 +5,133 @@
+        raise ValueError(_EMPTY_INPUT_ERROR_MESSAGE)
+
+    # Extract the sensitive features. If there are multiple sensitive features compress them into
+    # a single column which will serve as the binning vector.
+    sensitive_features_vector = _make_vector(sensitive_features,
+                                             _KW_SENSITIVE_FEATURES,
+                                             compress_multiple_columns=True)
 
     if enforce_binary_sensitive_feature:
-        unique_labels = np.unique(sensitive_features_vector)
-        if len(unique_labels) > 2:
-            raise RuntimeError("Sensitive features contain more than two unique values")
+        if len(np.unique(sensitive_features_vector)) > 2:
+            raise ValueError(_SENSITIVE_FEATURES_NON_BINARY_ERROR_MESSAGE)
 
-    # Extract the Y values
-    y_vector = _make_vector(y, "y")
+    if expect_y and enforce_binary_labels:
+        if not set(np.unique(y)).issubset(set([0, 1])):
+            raise ValueError(_LABELS_NOT_0_1_ERROR_MESSAGE)
 
     X_rows, _ = _get_matrix_shape(X, "X")
-    if X_rows != y_vector.shape[0]:
-        raise RuntimeError(_MESSAGE_X_Y_ROWS)
+
+    if expect_y:
+        y_vector = _make_vector(y, "y")
+        if X_rows != y_vector.shape[0]:
+            raise ValueError(_MESSAGE_X_Y_ROWS)
+    else:
+        y_vector = None
+
     if X_rows != sensitive_features_vector.shape[0]:
-        raise RuntimeError(_MESSAGE_X_SENSITIVE_ROWS)
+        raise ValueError(_MESSAGE_X_SENSITIVE_ROWS)
 
     return pd.DataFrame(X), y_vector, sensitive_features_vector
 
 
-def _make_vector(formless, formless_name):
+def _make_vector(formless, formless_name, compress_multiple_columns=False):
     formed_vector = None
     if isinstance(formless, list):
         formed_vector = pd.Series(formless)
     elif isinstance(formless, pd.DataFrame):
         if len(formless.columns) == 1:
             formed_vector = formless.iloc[:, 0]
+        elif compress_multiple_columns:
+            formed_vector = formless.apply(
+                lambda row: "-".join(
+                    [str(row[col]).replace("-", "_") for col in formless.columns]),
+                axis=1)
         else:
-            msgfmt = "{0} is a DataFrame with more than one column"
-            raise RuntimeError(msgfmt.format(formless_name))
+            raise ValueError(_MORE_THAN_ONE_COLUMN_ERROR_MESSAGE
+                             .format(formless_name, pd.DataFrame.__name__))
     elif isinstance(formless, pd.Series):
         formed_vector = formless
     elif isinstance(formless, np.ndarray):
         if len(formless.shape) == 1:
             formed_vector = pd.Series(formless)
         elif len(formless.shape) == 2 and formless.shape[1] == 1:
             formed_vector = pd.Series(formless[:, 0])
+        elif compress_multiple_columns:
+            formed_vector = np.apply_along_axis(
+                lambda row: "-".join([str(row[i]).replace("-", "_") for i in range(len(row))]),
```

Comment:
Could there be a safer character than `-` for joining the values?

Reply:
Maybe, but this will show up in the dashboard so I don't want # or $ or something weird, right? @rihorn2  @vingu @MiroDudik thoughts?

Reply:
Also, we're replacing those dashes first, so it's not like columns with dashes won't work

Reply:
see the comment above.

Reply:
changed to comma as pointed out by Miro